src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java

```diff
@@ -62,33 +65,36 @@ public void testExpiry() throws InterruptedException {
 	}
 	
 	private static class Counter {
-		private int counter;
+		private int ctr;
 		
 		public void increase() {
-			this.counter++;
+			this.ctr++;
 		}
 		public int getCounter() {
-			return this.counter;
+			return this.ctr;
 		}
 	}
 	
 	@Test
-	public void testAutoRefresh() throws InterruptedException {
+	public void testAutoRefresh() {
 		final Counter counter = new Counter();
-		
+		String testKey="key";
 		AutoRefreshingCacheMap<String, String> subject = new AutoRefreshingCacheMap<>("test", null, Duration.ofSeconds(10), Duration.ofMillis(500), key -> {
 			counter.increase();
 			return "refreshed";
 		});
-		
-		subject.put("key", "initial");
-		String value = subject.get("key");
+
+		subject.setRefresherThreadWithIncreasedPriority(true);
+		// Note that this also start the refresher thread immediately.
+		// It helps keeping this unit test to stay stable.
+
+		subject.put(testKey, "initial");
+		String value = subject.get(testKey);
 		Assert.assertEquals("initial", value);
 		
 		// we have to wait until the thread had a chance to refresh
-		Thread.sleep(1000);
-		
-		boolean exists = subject.containsKey("key");
+		await().atMost(5, SECONDS).until(() -> counter.getCounter()>0);
```

Comment:
I had a modified version of this test that did 500 iterations and timed for outliers. Only with these changes did we actually complete 500 iterations without failing the assert (ie not refreshing). Most outliers were 1001-1060ms on my desktop, but i have seen outliers of  +/- 1500ms.